### PR TITLE
feat(che-plugins): As part of automation, brings che-plugins.yaml file

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Brings che-plugins.yaml to define components that are neither VS Code Extensions/Theia plug-ins defined in che-theia-plugins.yaml or Editors components defined in che-editors.yaml

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18214#issuecomment-742396403

### How to test this PR?
For now the files are not used. (like current che-editors.yaml) but you may check yaml content is valid

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

Change-Id: I59bf1f056875a679ea9229884ad14b8fe953c7dd
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### CODE REVIEW

File: che-components.yaml [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+          preStop:
+            exec:
+            command: ['/bin/sh', '-c', '/scripts/backup.sh']
+        mountSources: true
+        ports:
+          - exposedPort: 4445
+        env:
+          - name: RSYNC_PORT
+            value: '2222'
+  - id: eclipse/che-machine-exec-plugin/nightly
```
Comment (svor) [inline che-components.yaml:96]: I think we should have similar format of separator between components (use empty line or not). As for me to have empty line between components is better

**Review by svor — APPROVED** [review]

File: che-components.yaml (hunk 1 above)
Comment (benoitf) [reply che-components.yaml:107]: @svor I've applied https://jsonformatter.org/yaml-formatter let me know if it is better now

Comment (svor) [reply che-components.yaml:107]: @benoitf yes, it's better now!

**Comment by benoitf** [comment]
After discussing with @davidfestal and @l0rd will name it che-plugins.yaml

**Pull request merged**